src/components/PaletteControls/PaletteControls.js

```diff
@@ -1,14 +1,58 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { setRandomHexcodes, toggleDisplayRandom } from '../../actions/index';
 
 export class PaletteControls extends Component {
+
+  findPaletteName = () => {
+    const  { palettes, currentPaletteId } = this.props;
+    const matchingPalette = palettes.find((palette) => {
+      return palette.id === currentPaletteId
+    })
+    return matchingPalette.name;
+  }
+
+  generateRandomColors = async () => {
+    let possibleHexValues = "0123456789ABCDEF";
+    let newColors = [];
+    for (let i = 0; i < 5; i++) {
+      let hexStringArray = [];
+      for (let j = 0; j < 6; j++) {
+        let generatedHexcode = possibleHexValues[(Math.floor(Math.random() * 16))]; 
+        hexStringArray.push(generatedHexcode);
+      }
+      let hexcode = hexStringArray.join('');
+      newColors.push(hexcode);
+    }
+    await this.props.setRandomHexcodes(newColors);
+    await this.props.toggleDisplayRandom(true);
+  }
+
   render() {
     return (
       <div className="PaletteControls">
-        <button className="generate-button">Generate New Palette!</button>
-        <h2 className="palette-header">PALETTE 1</h2>
+        {this.props.palettes.length ? (
+          <div className="palette-controls-container">
+            <button className="generate-button" onClick={this.generateRandomColors}>Generate New Palette!</button>
+            <h2 className="palette-header">{this.findPaletteName()}</h2>
+          </div>
+        ) : (
+          <p>loading...</p>
+        )}
```

Comment:
Maybe we could add a loading icon in here

Reply:
I love that idea!! 👍